python/pyarrow/tests/test_table.py

```diff
@@ -991,3 +991,60 @@ def test_table_function_unicode_schema():
     result = pa.table(d, schema=schema)
     assert result[0].chunk(0).equals(pa.array([1, 2, 3], type='int32'))
     assert result[1].chunk(0).equals(pa.array(['a', 'b', 'c'], type='string'))
+
+
+def test_table_subclassing():
+    class MyTable(pa.Table):
+        pass
+
+    table = MyTable.from_arrays([], [])
+    assert type(table) == MyTable
+
+    table = MyTable.from_pydict({})
+    assert type(table) == MyTable
+
+    schema = pa.schema([
+        pa.field('a', pa.int64()),
+        pa.field('b', pa.float64()),
+    ])
+    batch = pa.record_batch([pa.array([1]), pa.array([3.14])],
+                            names=['a', 'b'])
+    table = MyTable.from_batches([batch], schema)
+    assert type(table) == MyTable
+
+    import pandas as pd
```

Comment:
Use `@pytest.mark.pandas` if a test requires Pandas, as it should be possible to run the test suite without Pandas installed.